internal/rbd/rbd_util.go

```diff
@@ -819,6 +822,15 @@ func genVolFromVolumeOptions(ctx context.Context, volOptions, credentials map[st
 		rbdVol.Mounter = rbdDefaultMounter
 	}
 
+	// if the KMS is of type VaultToken, additional metadata is needed
+	// depending on the tenant, the KMS can be configured with other
+	// options
+	// FIXME: this works only on Kubernetes, how do other CO supply metadata?
+	rbdVol.Owner, ok = volOptions["csi.storage.k8s.io/pvc/namespace"]
```

Comment:
Is this needed only when encryption is enabled? if this is not passed we are going to get Debug logs for all volume create requests.

Reply:
It is not required (at the moment) for non-encrypted volumes. The option to pass extra metadata is passed with all deployments, as we can not  know in advance if encryption will be used, or not. So the owner will be available everywhere.

In production environments, debug logging should not be enabled. I'd expect only errors (and higher) to be logged in production environments by default.

Reply:
in storageclass "encrypted" will be set to true right based on that we can do this operation?

Reply:
We could, but in the end it is just an other attribute of the PVC and RBD-image. It does not hurt to have it (and makes testing of the rados key/value simpler).